### ISSUE TITLE
[ts_proto_library] - Editor support 

### ISSUE THREAD

**Issue by Toxicable** [bazelbuild/rules_typescript#368]
# 🚀 feature request

### Relevant Rules
`ts_proto_library`

### Description
Currently `ts_proto_library` produces `.js` generated code and `.d.ts` files to match those, which works well from the perspective of the compiler but for devs it makes for a bad experience since your editor will highlight those imports as errors because you're importing files which don't exist at dev time. This means you loose all type support for any of the apis and interfaces used from your proto files

### Describe the solution you'd like
`ts_proto_library` should generate `.d.ts` files into your source directory instead of output so they can be used at dev time.

### Describe alternatives you've considered
An alternative would be to manually write interfaces to match the proto files you've written, but then whats the point of using it in the first place?

**Comment by ashi009** [comment]
This can be done through path mapping (https://github.com/bazelbuild/rules_typescript#writing-typescript-code-for-bazel). In case you are not using non-relative imports, add `bazel-bin` to rootDirs next to `.`. Which will make your ts-server to pick up those generated files.

IMHO, Generating code to the source tree is generally a bad idea, as we should aim to have a single source of truth when possible, mixing generated files with the source will make future maintenance difficult.

**Comment by Toxicable** [comment]
I see your point about generated code, you're right, we should rely on the .proto, not the generated code.

While using your proposed solution of adding `bazel-out/host/bin` into `rootDirs` would work I suspect that it wont scale.
Doing this means that the TS language will pick up at minimum 2x of every source file  - 1 - the actual source file, 2 - the .d.ts's generated by ts_libraray.
Then if you're using `ng_module` it's at least 4x of each file - 2 as above, 3 - ngfactory, 4 - ngsummary
Having to read and parse so many files will quickly bring tsserver to a crawl and language support within the editor down with it.

**Comment by ashi009** [comment]
You are right about the numbers.  However, the conclusion here is a bit off. The problem here is that ts language server itself is not designed to scale, as it doesn't load files in a heuristic way and use a sensible cache strategy. It will fail eventually when your repo grows, regardless of using the path mapping or not.

We have O(1k) ts files with O(1k) generated files. So far, ts server can cope with that, as once the symbols are loaded it will only do the incremental patches to the indices in the memory. However, the initial bootstrap time will take a while.
